R/hp_datatree.R

```diff
@@ -0,0 +1,141 @@
+dt_root <- function(x) {
+  names(x$Get("level", filterFun = data.tree::isRoot))
+}
+
+dt_root_df <- function(x) {
+  x <- dt_root(x)
+  data.frame(
+    from = x,
+    to = x,
+    stringsAsFactors = FALSE
+  )
+}
+
+dt_collect <- function(x) {
+  requireNamespaceQuietStop("data.tree")
+  source <- data.tree::ToDataFrameNetwork(x)
+  if (length(x$fieldsAll) == 0) {
+    source
+  } else {
+    out <- lapply(x$fieldsAll, function(attribute) {
+      data.tree::ToDataFrameNetwork(x, attribute)[3]
+    })
+
+    vctrs::vec_cbind(source, do.call(vctrs::vec_cbind, out))
+
+  }
+}
+
+#' Create a hierplane object using data.tree
+#'
+#' The `data.tree` package is a popular, general purpose hierarchical data
+#' structure for R. Therefore, `hp_datatree` tries to make it so hierplane is
+#' compatible with `data.tree` objects.
+#'
+#' @param .data A `data.tree` object of class "Node".
+#' @param title A title, defaults to "Hierplane", this serves as the header/title of the hierplane.
+#' @param attributes Attributes to assign to the nodes, these are the annotation in the nodes.
```

Comment:
i think we need to comb through the definition of these fields from all the functions and make sure they are defined the same. for example i think attributes is defined as "label in the nodes" in another function.

Reply:
Yeah I agree... since this isn't a super function heavy package. I say we revisit once we feel things are stable.